### PR TITLE
Awesome IPLD Endeavour

### PR DESCRIPTION
**Pull request by diasdavid**
- [x] remove extension support (no need with CID)
- [x] blockstore starts getting blockBlobs instead of blocks (so that it can know the right key)

### CODE REVIEW

**Comment by dignifiedquire** [comment]
Why not add cid support directly to Block, The current `{block, cid}` constructs seems akward and easy to break accidentally.

**Comment by diasdavid** [comment]
I scratched my head a lot thinking about this. Although having a .cid() on the block would be super sweet, you can't really tell which hash function to use, otherwise, it breaks the Block abstraction.

Only the Block Service, hinted by the caller with the CID, can tell which data type and hash function should it use for the data it is looking for.

File: package.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "name": "ipfs-repo",
   "version": "0.9.1",
   "description": "IPFS Repo implementation",
-  "main": "lib/index.js",
+  "main": "src/index.js",
```
Comment (dignifiedquire) [inline package.json]: TODO: change before merge

File: src/stores/blockstore.js [hunk 2]
```diff
@@ -5,16 +5,16 @@ const pull = require('pull-stream')
 const Lock = require('lock')
 const base32 = require('base32.js')
 const path = require('path')
-const write = require('pull-write')
+const pullWrite = require('pull-write')
```
Comment (dignifiedquire) [inline src/stores/blockstore.js]: why?

File: src/stores/blockstore.js [hunk 3]
```diff
@@ -5,16 +5,16 @@ const pull = require('pull-stream')
...
+const pullDefer = require('pull-defer/source')
 
 const PREFIX_LENGTH = 5
 
 exports = module.exports
 
-function multihashToPath (multihash, extension) {
-  extension = extension || 'data'
+function multihashToPath (multihash) {
+  const extension = 'data'
```
Comment (dignifiedquire) [inline src/stores/blockstore.js]: if this is fixed, why not make it a constant like `PREFIX_LENGTH`

File: src/stores/blockstore.js [hunk 4]
```diff
@@ -27,82 +27,113 @@ exports.setUp = (basePath, BlobStore, locks) => {
...
-
-            written.push(meta)
-            cb()
-          })
-        }), cb)
+      const sink = pullWrite((blockBlobs, cb) => {
+        const tasks = blockBlobs.map((blockBlob) => {
+          return (cb) => {
+            writeBlock(blockBlob, (err, meta) => {
+              if (err) {
```
Comment (dignifiedquire) [inline src/stores/blockstore.js]: the nesting is strong in this one,  maybe simplify this somehow?

File: src/stores/locks.js [hunk 5]
```diff
@@ -37,16 +41,22 @@ exports.setUp = (basePath, BlobStore) => {
...
+              return cb(err)
+            }
+
+            if (exists) {
+              return cb(new Error('failed to remove lock'))
+            }
+
+            cb()
+          })
+        }
```
Comment (dignifiedquire) [inline src/stores/locks.js]: :-1: this change makes it harder to read, and is a matter of personal preference

File: src/stores/blockstore.js [hunk 6]
```diff
@@ -27,82 +27,113 @@ exports.setUp = (basePath, BlobStore, locks) => {
...
+        parallel(tasks, cb)
       }, null, 100, (err) => {
         ended = err || true
-        if (push) push(ended)
+        if (push) {
+          push(ended)
+        }
       })
 
+      // TODO ??Why does a putStream need to be a source as well??
```
Comment (diasdavid) [inline src/stores/blockstore.js]: @dignifiedquire could you clarify on this one?

File: src/stores/blockstore.js [hunk 7]
```diff
@@ -27,82 +27,94 @@ exports.setUp = (basePath, BlobStore, locks) => {
...
+        parallel(tasks, cb)
       }, null, 100, (err) => {
         ended = err || true
-        if (push) push(ended)
+        if (push) {
+          push(ended)
+        }
       })
 
+      // TODO ?? Why does a putStream need to be a source as well??
```
Comment (diasdavid) [inline src/stores/blockstore.js]: @dignifiedquire

**Comment by diasdavid** [comment]
for record, @dignifiedquire approved

**Pull request merged**